command/filemanager.go

```diff
@@ -10,70 +10,49 @@ import (
 	"runtime"
 	"strings"
 
-	"github.com/DataDog/test-infra-definitions/common/utils"
 	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 )
 
-const (
-	linuxTempDir = "/tmp"
-)
-
 type FileManager struct {
```

Comment:
Indeed but I already spent too many time fixing `FileManager`. Feel free to update the code in a later PR.

Reply:
> I'm not sure why FileManager is not an interface

I think it is better that `FileManager` is a struct and takes an interface like `osCommand` which is specific to the operating system. The reason is that you may want to not expose directly the low level function of `osCommand`
```go
func (fm *FileManager) MyFunction() {
        // Do something independent of the OS
       fm.command.MyLowLevelFunction()
       // Do something else independent of the OS
}
```

Note: I have realised that `CreateTemporaryDirectory` shouldn't be in `osCommand` and `osCommand` should only provide a `TemporaryDirectory`.

In addition, in Go, it is not straightforward to implement the pattern for `MyFunction` with a `WindowsFileManager` and `UnixFileManager`. Indeed, in order to factorize the common code, `UnixFileManager` would need to have an embedded struct that provide `MyFunction`. The embedded struct would require an interface to access `MyLowLevelFunction` which would be implemented by `UnixFileManager`. There is a loop in the initialisation of `UnixFileManager` and the embedded struct.

> I'm not sure that putting a mix of everything in *OSCommand makes sense

If you follow the [Interface segregation principle](https://en.wikipedia.org/wiki/Interface_segregation_principle) from  [SOLID](https://en.wikipedia.org/wiki/SOLID) principle,  `FileManager` should only require the methods it uses, ie, everything except `BuildCommandString`. To do so, you could have a `osCommand` that provides a `Filesystem` interface and a `Process` interface for example. 

I think this is not needed for now because:
- There are not many functions and don't expect to grow that much 
- It would be easy to change the design later
- Changing  the design now or later will cost nearly the same
- If there are no new methods require, the current code is simple [KISS](https://en.wikipedia.org/wiki/KISS_principle).

If you have a strong opinion that `OSCommand` should be split let me know.